src/volatile.rs

```diff
@@ -3,6 +3,12 @@
 #[repr(transparent)]
 pub struct ReadOnly<T: Copy>(T);
 
+impl<T: Debug + Copy> Debug for ReadOnly<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.debug_tuple("ReadOnly").field(&self.0).finish()
```

Comment:
`ReadOnly` is intended to wrap MMIO registers which may have side-effects when reading, so it's probably not a good idea to implement `Debug`, as adding debug output may change behaviour. Also any reading should be done with `read_volatile`.

Reply:
Indeed, you are right, I didn't consider the issue of side effects.